prov/verbs/src/verbs_info.c

```diff
@@ -518,11 +519,26 @@ static int fi_ibv_get_device_attrs(struct ibv_context *ctx,
 	if (ret)
 		return ret;
 
-	ret = ibv_query_port(ctx, 1, &port_attr);
-	if (ret) {
-		VERBS_INFO_ERRNO(FI_LOG_FABRIC,
-				 "ibv_query_port", errno);
-		return -errno;
+	for (i = 1; i < 32; i++) {
+		ret = ibv_query_port(ctx, i, &port_attr);
+		if (ret) {
+			VERBS_INFO_ERRNO(FI_LOG_FABRIC,
+					 "ibv_query_port", errno);
+			return -errno;
+		}
+		if (port_attr.state == IBV_PORT_ACTIVE) {
+			port_num = i;
+			break;
+		}
+	}
+
+	if (!port_num) {
```

Comment:
nit: you can remove port_num, and compare i == 32, with any other value i is the port num

Reply:
Thanks, good catch!

Reply:
@shefty Not sure why, but the changes from [my fork](https://github.com/gladkovdmitry17/libfabric/commit/c09924b361bd4edc4fc3c2e9094a431befe05c8b) isn't synced with the PR

Reply:
@shefty Fixed. Now it works :)